hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -2580,20 +2580,27 @@ public KeyProvider call() throws Exception {
 
   @Override
   public OzoneFsServerDefaults getServerDefaults() throws IOException {
+    if (omVersion.compareTo(OzoneManagerVersion.SERVER_DEFAULTS) < 0) {
+      return null;
+    }
     long now = Time.monotonicNow();
     if ((serverDefaults == null) ||
         (now - serverDefaultsLastUpdate > serverDefaultsValidityPeriod)) {
-      serverDefaults = ozoneManagerClient.getServerDefaults();
-      serverDefaultsLastUpdate = now;
+      try {
+        serverDefaults = ozoneManagerClient.getServerDefaults();
```

Comment:
We discussed this in the US community call today. Instead of adding a new RPC, we can probably add whatever information is required to initial service info call to the OM. The RPC client can save it in memory and return it here. Future invocations based on time can issue another get service info call. This approach will not require a new OM component version.

Reply:
Thanks for the input @errose28. I was able to use service info to return server defaults.